packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -16,253 +16,17 @@ import 'ticker_provider.dart';
-      return;
-    }
-
-    overlay._entries.remove(this);
-    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
-      SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
-        overlay._markDirty();
-      });
-    } else {
-      overlay._markDirty();
-    }
-  }
-
-  /// Cause this entry to rebuild during the next pipeline flush.
-  ///
-  /// You need to call this function if the output of [builder] has changed.
-  void markNeedsBuild() {
-    assert(!_disposedByOwner);
-    _key.currentState?._markNeedsBuild();
-  }
-
-  void _didUnmount() {
-    assert(!mounted);
-    if (_disposedByOwner) {
-      _overlayStateMounted.dispose();
-    }
-  }
-
-  bool _disposedByOwner = false;
-
-  /// Discards any resources used by this [OverlayEntry].
-  ///
-  /// This method must be called after [remove] if the [OverlayEntry] is
-  /// inserted into an [Overlay].
-  ///
-  /// After this is called, the object is not in a usable state and should be
-  /// discarded (calls to [addListener] will throw after the object is disposed).
-  /// However, the listeners registered may not be immediately released until
-  /// the widget built using this [OverlayEntry] is unmounted from the widget
-  /// tree.
-  ///
-  /// This method should only be called by the object's owner.
-  void dispose() {
-    assert(!_disposedByOwner);
-    assert(_overlay == null, 'An OverlayEntry must first be removed from the Overlay before dispose is called.');
-    _disposedByOwner = true;
-    if (!mounted) {
-      _overlayStateMounted.dispose();
-    }
-  }
-
-  @override
-  String toString() => '${describeIdentity(this)}(opaque: $opaque; maintainState: $maintainState)';
-}
-
-class _OverlayEntryWidget extends StatefulWidget {
-  const _OverlayEntryWidget({
-    required Key key,
-    required this.entry,
-    this.tickerEnabled = true,
-  }) : assert(key != null),
-       assert(entry != null),
-       assert(tickerEnabled != null),
-       super(key: key);
-
-  final OverlayEntry entry;
-  final bool tickerEnabled;
-
-  @override
-  _OverlayEntryWidgetState createState() => _OverlayEntryWidgetState();
-}
-
-class _OverlayEntryWidgetState extends State<_OverlayEntryWidget> {
-  @override
-  void initState() {
-    super.initState();
-    widget.entry._overlayStateMounted.value = true;
-  }
-
-  @override
-  void dispose() {
-    widget.entry._overlayStateMounted.value = false;
-    widget.entry._didUnmount();
-    super.dispose();
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    return TickerMode(
-      enabled: widget.tickerEnabled,
-      child: widget.entry.builder(context),
-    );
-  }
-
-  void _markNeedsBuild() {
-    setState(() { /* the state that changed is in the builder */ });
-  }
+/// Each invocation of one such function returns the next element in the
+/// collection. When the collection is exhausted the function returns null.
+typedef _Iterator<T extends Object> = T? Function();
```

Comment:
Plain old `Iterable` is not good enough?

Reply:
Well I found it a bit awkward to use `Iterable` or `Iterator` for this. This class can't have 2 implementations for paint order traversal and hit test order traversal so we'd have to introduce more new classes.